### PR TITLE
test(coverage): Add babel-istanbul for coverage

### PR DESCRIPTION
**Pull request by Jerskouille**
Linked to  #355 .

This adds `npm run test:coverage` which uses `babel-istanbul`.
The difference between `istanbul` and `babel-istanbul` is that it shows coverage on the code before it is compiled by `babel`.

### CODE REVIEW

**Comment by vvo** [comment]
By analyzing the blame information on this pull request, we identified @vvo, @bobylito and @redox to be potential reviewers

**Comment by redox** [comment]
@Jerskouille could you also add the associated badge?

**Comment by vvo** [comment]
I am a bot! I will fix this

File: package.json [hunk 1]
```diff
@@ -14,8 +14,9 @@
     "lint": "eslint . --quiet --no-color && sass-lint --verbose",
     "release": "./scripts/release.sh",
     "release-beta": "./scripts/release-beta.sh",
-    "test": "BABEL_ENV=test mocha --opts scripts/mocha.opts --reporter dot && npm run lint",
-    "test:watch": "BABEL_ENV=test mocha --opts scripts/mocha.opts --reporter min --watch",
+    "test": "BABEL_ENV=test mocha --opts scripts/mocha.opts --compilers js:babel/register --reporter dot && npm run lint",
+    "test:coverage": "BABEL_ENV=test babel-node node_modules/.bin/babel-istanbul cover node_modules/.bin/_mocha -- --opts scripts/mocha.opts --reporter dot && npm run lint",
```
Comment (vvo) [inline package.json]: You can use "babel-istanbul" without specifying nodes_modules/.bin.. It works

**Comment by vvo** [comment]
Good first steps, thanks! Do you already know how we can automate it and use any service (jenkings, coveralls..?) so that we track the coverage a little (badge, PR)?

File: package.json (hunk 1 above)
Comment (Jerskouille) [inline package.json]: Not with `babel-node` that is needed for it to work. (Trust me, I've tried reducing the commands as much as possible)
Here is the error when I try without it:
```txt
Error: Cannot find module 'path_to_my_project/babel-istanbul'

```

**Comment by Jerskouille** [comment]
I have no knowledge of coveralls. I've started looking, but I am not confident giving `coveralls` access to all the algolia repositories when I try to login with GitHub.

**Comment by redox** [comment]
> I have no knowledge of coveralls. I've started looking, but I am not confident giving coveralls access to all the algolia repositories when I try to login with GitHub.

We're also using coveralls on a bunch of projects, don't worry ;)

File: package.json (hunk 1 above)
Comment (vvo) [inline package.json]: Ok thanks for clarifications

**Comment by vvo** [comment]
Ok let's move forward and add coverall etc in another PR since this one is "linked" and already useful

**Comment by pixelastic** [comment]
I think coverall is using their own tools to calculate the coverage automatically, we have nothing to add (except if we want to get the value before pushing to coverall)

**Pull request merged**